grammar.js

```diff
@@ -1,157 +1,165 @@
-const PREC = {
-  PATH: 10, // foo/bar
-  BIND: 5, // foo:bar
-};
-
-const SYMBOL_HEAD =
-  /[^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:.0-9]/;
-
-const SYMBOL_TAIL =
-  choice(SYMBOL_HEAD, /[.0-9]/);
-
-const SYMBOL =
-  token(seq(SYMBOL_HEAD, repeat(SYMBOL_TAIL)));
-
-const SUBPATH =
-  repeat1(seq("/", optional(SYMBOL)));
+/**
+ * @file Bass grammar for tree-sitter
+ * @author Alex Suraci <suraci.alex@gmail.com>
+ * @author Amaan Qureshi <amaanq12@gmail.com>
+ * @license MIT
+ * @see {@link https://bass-lang.org/|official website}
+ */
+
+/* eslint-disable arrow-parens */
+/* eslint-disable camelcase */
+/* eslint-disable-next-line spaced-comment */
+/// <reference types="tree-sitter-cli/dsl" />
+// @ts-check
+
+const SYMBOL = token(
+  /[^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:.0-9][^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:.0-9.]*/,
+);
+
+const PATH = token(choice(
+  /\.\/[A-Za-z0-9\.\/]*/,
+  /\/[A-Za-z0-9\.\/]*/,
+  /[A-Za-z0-9\.\/]*\/[A-Za-z0-9\.\/]*/,
+));
 
 module.exports = grammar({
   name: 'bass',
 
-  rules: {
-    source: $ =>
-      repeat(choice($._form, $._gap)),
+  extras: $ => [
+    $.comment,
+    /[\s,]/,
+  ],
 
-    _gap: $ =>
-      choice($._ws, $.comment),
+  supertypes: $ => [
+    $.form,
+    $.literal,
+  ],
 
-    _ws: _ =>
-      // Latin-1 spaces should cover it in practice, + comma
-      token(repeat1(/[\t\n\v\f\r \u0085\u00A0,]/)),
+  word: $ => $.symbol,
 
-    comment: _ =>
-      token(/(;|#!).*\n?/),
+  rules: {
+    source: $ => repeat($.form),
 
-    _form: $ =>
+    form: $ =>
       choice(
-        // scalars
-        $.int,
-        $.string,
-        $.ignore,
-        $.null,
-        $.bool,
+        // literals
+        $.literal,
+
+        // Identifier
         $.symbol,
+
         $.keyword,
+
         // collections
         $.list,
         $.cons,
         $.scope,
+
         // paths
-        $.command,
+        $._module,
         $.path,
         $.symbind,
+
         // meta
         $.meta,
       ),
 
-    int: _ =>
-      token(prec(10, seq(optional(/[+-]/), repeat1(/[0-9]/)))),
+    keyword: $ => seq(':', choice($.symbol, $.path, $._module)),
```

Comment:
I don't think this would parse correctly:

```bass
git:github/vito/bass/ref/main/
```

It should parse as:

* `git` - symbol
* `:github` - keyword
* `/vito/bass/ref/main/` - path

Note: the above example is syntax sugar for `(((((:github git) ./vito/) ./bass/) ./ref/) ./main/)`:

```clojure
=> git:github/vito/bass/ref/main/
=> resolve image config for docker.io/alpine/git@sha256:66b210a97b
=> docker-image://docker.io/alpine/git@sha256:66b210a97bc07bfd4019
-> resolve docker.io/alpine/git@sha256:66b210a97bc07bfd4019826bcd1
=> git ls-remote https://github.com/vito/bass main [0.48s]
=> exporting to client directory [0.00s]
-> copying files 83B [0.00s]
<thunk EJM1CCLPNKLGC: (.git)>/
=> (((((:github git) ./vito/) ./bass/) ./ref/) ./main/)
=> resolve image config for docker.io/alpine/git@sha256:66b210a97b
=> docker-image://docker.io/alpine/git@sha256:66b210a97bc07bfd4019
-> resolve docker.io/alpine/git@sha256:66b210a97bc07bfd4019826bcd1
=> git ls-remote https://github.com/vito/bass main [0.53s]
=> exporting to client directory [0.00s]
-> copying files 83B [0.00s]
<thunk EJM1CCLPNKLGC: (.git)>/
```